File: leetcode/go/_001/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(twoSum([]int{3, 2, 4}, 6))
}

func twoSum(nums []int, target int) []int {
	mp := make(map[int]int)
	for i, v := range nums {
		mp[v] = i
	}

	for i, num := range nums {
		t := target - num
		if v, ok := mp[t]; ok && v != i {
			return []int{i, v}
		}
	}

	return []int{}
}

File: leetcode/python/_001/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = dict()
        for i in range(len(nums)):
            val = target - nums[i]
            if val in mp:
                return [mp[val], i]
            mp[nums[i]] = i
        return []

print(Solution().twoSum([2, 7, 11, 15], 9))
print(Solution().twoSum([3, 2, 4], 6))
print(Solution().twoSum([3, 3], 6))